### PR TITLE
[google_maps_flutter_web] Allow Markers with icon:null

### PR DESCRIPTION
**Pull request by ditman**
## Description

The current conversion of our `Marker` type to `gmaps.Marker` assumes that `marker.icon` is never null ([internally, it defaults to `BitmapDescriptor.defaultMarker`](https://github.com/flutter/plugins/blob/master/packages/google_maps_flutter/google_maps_flutter_platform_interface/lib/src/types/marker.dart#L165)), however users can break the default by creating a marker like so:

```dart
Marker(markerId: MarkerId('boom'), icon: null);
```

This PR makes the `Marker` to `gmaps.Marker` conversion resilient to the case where the passed `icon` is `null`.

## Related Issues

* Fixes https://github.com/flutter/flutter/issues/64938

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [x] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart [hunk 1]
```diff
@@ -379,22 +379,24 @@ gmaps.MarkerOptions _markerOptionsFromMarker(
...
 ) {
-  final iconConfig = marker.icon.toJson() as List;
+  final iconConfig = marker.icon?.toJson() as List;
   gmaps.Icon icon;
 
-  if (iconConfig[0] == 'fromAssetImage') {
-    // iconConfig[2] contains the DPIs of the screen, but that information is
-    // already encoded in the iconConfig[1]
+  if (iconConfig != null) {
+    if (iconConfig[0] == 'fromAssetImage') {
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:386]: nits:
Add `assert(iconConfig.length >= 2)`

**Review by cyanglaz — APPROVED** [review]
LGTM % nits.

File: packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md [hunk 2]
```diff
@@ -1,3 +1,7 @@
+## 0.1.1
```
Comment (cyanglaz) [inline packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md:1]: Does it have a public API change, do we need to bump the minor version? Maybe just 0.1.0+2?

Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_web/CHANGELOG.md:1]: Yeah, this is not a public api change, I'll try to bump +1... I still don't understand the rules of the validation script to let me use +1s or not!

File: packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart (hunk 1 above)
Comment (ditman) [reply packages/google_maps_flutter/google_maps_flutter_web/lib/src/convert.dart:386]: I have to add this inside of the iconConfig[0] == 'fromAssetImage', other iconConfigs might not be as long (like the default icon)

**Comment by ditman** [comment]
Addressed PR comments, and rebased branch.

**Pull request merged**